mongo/options/clientoptions.go

```diff
@@ -279,26 +279,32 @@ func (c *ClientOptions) ApplyURI(uri string) *ClientOptions {
 			tlsConfig.InsecureSkipVerify = true
 		}
 
+		var x509Subject string
+		var keyPasswd string
+		if cs.SSLClientCertificateKeyPasswordSet && cs.SSLClientCertificateKeyPassword != nil {
+			keyPasswd = cs.SSLClientCertificateKeyPassword()
+		}
 		if cs.SSLClientCertificateKeyFileSet {
-			var keyPasswd string
-			if cs.SSLClientCertificateKeyPasswordSet && cs.SSLClientCertificateKeyPassword != nil {
-				keyPasswd = cs.SSLClientCertificateKeyPassword()
-			}
-			s, err := addClientCertFromFile(tlsConfig, cs.SSLClientCertificateKeyFile, keyPasswd)
-			if err != nil {
-				c.err = err
-				return c
-			}
+			x509Subject, err = addClientCertFromConcatenatedFile(tlsConfig, cs.SSLClientCertificateKeyFile, keyPasswd)
+		} else if cs.SSLCertificateFileSet || cs.SSLPrivateKeyFileSet {
```

Comment:
I would have expected that both of these would be set so that you could only check one, but that got me to realize that this PR only addresses the connection string.  What about SSLOpt?  I think anything possible in a connection string needs to be possible from code.  Given that you have to validate having both cert/key, I think that also means SSLOpt needs to gain builder functions to guide users (and still probably needs checking validity on use or in `SetSSL`.